### PR TITLE
feature/back-615 - `Dystopia` exchange integration

### PR DESCRIPTION
**Pull request by Verisana**
Integration from [PR#108](https://github.com/paraswap/paraswap-dex-lib/pull/108)

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -66,3 +66,5 @@ export const AugustusAddress: { [nid: number]: Address } = {
   [Network.AVALANCHE]: '0xDEF171Fe48CF0115B1d80b88dc8eAB59176FEe57',
   [Network.FANTOM]: '0xDEF171Fe48CF0115B1d80b88dc8eAB59176FEe57',
 };
+
+export const subgraphTimeout = 20 * 1000;
```
Comment (erez-paraswap) [inline src/constants.ts:70]: Special reason to have a constant starts in lowercase? Should be ALL_UPPERCASE

File: tests/constants-e2e.ts [hunk 2]
```diff
@@ -535,7 +535,7 @@ export const Holders: {
   [Network.POLYGON]: {
     MATIC: '0xe7804c37c13166fF0b37F5aE0BB07A3aEbb6e245',
     DAI: '0x06959153B974D0D5fDfd87D561db6d8d4FA0bb0B',
-    WETH: '0xd3d176F7e4b43C70a68466949F6C64F06Ce75BB9',
+    WETH: '0x72a53cdbbcc1b9efa39c834a540550e23463aacb',
```
Comment (erez-paraswap) [inline tests/constants-e2e.ts:538]: Is this intended change? It's so radical. Have you ran all tests to see all of them pass?

File: tsconfig.json [hunk 3]
```diff
@@ -18,6 +18,6 @@
     "typeRoots": ["node_modules/@types"],
     "esModuleInterop": true
   },
-  "include": ["src/**/*"],
+  "include": ["src/**/*", "tests/**/*"],
```
Comment (erez-paraswap) [inline tsconfig.json:21]: now we include tests/** in build/dist? Why is it changed?

**Review by erez-paraswap — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 3 above)
Comment (Verisana) [reply tsconfig.json:21]: I think I can remove it. It was done by @bogdoslavik because he has issues with IDE. After this we have:
```
"exclude": ["**/*.test.ts"]
```
So, it would add only test utils to build, though we don't need them.
Fixed

File: tests/constants-e2e.ts (hunk 2 above)
Comment (Verisana) [reply tests/constants-e2e.ts:538]: So, the reason to change was because some test were failing :)

File: tsconfig.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
     "module": "commonjs",
     "noImplicitAny": true,
     "outDir": "build",
-    "rootDir": "src",
+    "rootDirs": ["src"],
```
Comment (ColonelJ) [inline tsconfig.json:8]: Why this change?  `rootDir` and `rootDirs` are different things, and I'm not sure how it affects the build output...

File: src/dex/balancer-v2/balancer-v2.ts [hunk 5]
```diff
@@ -18,26 +18,25 @@ import {
   MAX_INT,
   MAX_UINT,
   Network,
+  subgraphTimeout,
```
Comment (ColonelJ) [inline src/dex/balancer-v2/balancer-v2.ts:21]: I think this was renamed in `constants.ts`, but not here and other files.

Comment (Verisana) [reply src/dex/balancer-v2/balancer-v2.ts]: Fixed

File: tsconfig.json (hunk 4 above)
Comment (Verisana) [reply tsconfig.json]: Fixed to previous state

File: src/dex/uniswap-v2/constants.ts [hunk 6]
```diff
@@ -15,6 +16,7 @@ export const AllUniswapForks = [
   ...Object.keys(BiSwapConfig),
   ...Object.keys(DfynConfig),
   ...Object.keys(ExcaliburConfig),
+  ...Object.keys(DystopiaConfig),
```
Comment (ColonelJ) [inline src/dex/uniswap-v2/constants.ts:19]: I think Dystopia should not be included in `AllUniswapForks` nor `UniswapForksWithNetwork` because it is not a standard fork, and it will break things if it gets mixed with another fork as `UniswapForkOptimized`.  Also, it doesn't support buy.  Comment it out on both and include a note in the comment above which currently covers `BakerySwap`.

**Review by ColonelJ — CHANGES_REQUESTED** [review]

File: src/dex/uniswap-v2/constants.ts (hunk 6 above)
Comment (Verisana) [reply src/dex/uniswap-v2/constants.ts]: Fixed

**Review by ColonelJ — APPROVED** [review]

**Review by ColonelJ — CHANGES_REQUESTED** [review]
I see very wrong prices in staging, something is wrong with stable pool price calculations.

**Review by ColonelJ — APPROVED** [review]
